CMakeLists.txt

```diff
@@ -210,6 +213,7 @@ if(USE_CUDA)
     function(add_histogram hsize hname hadd hconst hdir)
       add_library(histo${hsize}${hname} OBJECT src/treelearner/kernels/histogram${hsize}.cu)
       set_target_properties(histo${hsize}${hname} PROPERTIES CUDA_SEPARABLE_COMPILATION ON)
+      set_target_properties(histo${hsize}${hname} PROPERTIES CUDA_ARCHITECTURES "${CUDA_ARCHS_STRING}")
```

Comment:
Do you have any thought about how is this property compatible with following two lines
https://github.com/microsoft/LightGBM/blob/adf36d766afb6261276d0624f4f5b216d51880c8/CMakeLists.txt#L177
https://github.com/microsoft/LightGBM/blob/adf36d766afb6261276d0624f4f5b216d51880c8/CMakeLists.txt#L179
where we manually append target architectures taken from `CUDA_SELECT_NVCC_ARCH_FLAGS` function to compiler flags?
https://cmake.org/cmake/help/latest/module/FindCUDA.html#commands

Reply:
Maybe it will be better to simply set `CUDA_ARCHITECTURES` to `OFF` as it is shown in the last example?
https://cmake.org/cmake/help/latest/policy/CMP0104.html#examples
I believe by doing so we will preserve the old behavior (setting architectures manually in the compiler flags) and in in consequence compatibility with CMake < 3.18 and make the warning disappear in CMake >= 3.18. Please let me know WDYT about this.

Reply:
Hmmm, I agree with that. Since already manually append target arch into CMAKE_CUDA_FLAGS, we can just turn off `CUDA_ARCHITECTURES` to keep this change the simplest.